tools/sof-ipc-timer.py

```diff
@@ -0,0 +1,149 @@
+            usecs = secs * 1000000 + dt_object.microsecond
+            span_end_pos = match_obj.span()[1]
+            # msg_type shoule be either "     ", "reply", or "done "
+            msg_type = line[match_obj.span()[0] + 8 : match_obj.span()[1]]
+            msg_part = line[span_end_pos:].split()
+            primary = int(msg_part[1].split('|')[0], 16)
+            extension =  int(msg_part[1].split('|')[1].rstrip(":"), 16)
+            if msg_part[2] == "MOD_INIT_INSTANCE" or msg_part[2] == "MOD_LARGE_CONFIG_SET":
+                if msg_type == "     ":
+                    mod_id = primary & 0xFFFFFF
+                    if mod_id == 0:
+                        mod_id = None
+                        continue
+                    start = usecs
+                if mod_id == None:
+                    continue
+                comp = comp_data[mod_id]
+                if msg_type == "reply" and args.reply_timings:
+                    if msg_part[2] == "MOD_INIT_INSTANCE":
+                        print("%s:\tinit reply\t%d us" % (comp.wname, usecs - start))
+                    elif msg_part[2] == "MOD_LARGE_CONFIG_SET":
+                        print("%s:\tconf reply\t%d us" % (comp.wname, usecs - start))
+                    start = usecs
+                elif msg_type == "done ":
+                    module_id = ""
+                    pipeline_id = ""
+                    if args.module_id:
+                        module_id = "\tmodule id: " + format(mod_id, '#08x')
+                    if args.pipeline_id:
+                        pipeline_id = "\tpipeline id: " + str(comp.ppln_id)
+                    if msg_part[2] == "MOD_INIT_INSTANCE":
+                        print("%s:\tinit done\t%d us%s%s" %
+			      (comp.wname, usecs - start, module_id, pipeline_id))
+                    elif msg_part[2] == "MOD_LARGE_CONFIG_SET":
+                        print("%s:\tconf done\t%d us%s%s" %
+			      (comp.wname, usecs - start, module_id, pipeline_id))
+                    mod_id = None
+                    start = None
+                    end = None
+
+def parse_args():
+    '''Parse command line arguments'''
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                     description=__doc__)
+    parser.add_argument('filename')
+    parser.add_argument('-r', '--reply-timings', action="store_true", default=False,
+                        help='Show time to reply message, "done" time is from "reply"')
+    parser.add_argument('-m', '--module-id', action="store_true", default=False,
+                        help='Show module id')
+    parser.add_argument('-p', '--pipeline-id', action="store_true", default=False,
+                        help='Show pipeline id')
+
+    return parser.parse_args()
+
+def main():
+    args = parse_args()
+
+    with open(args.filename, 'r', encoding='utf8') as file:
```

Comment:
IMHO a tool like this should really be "stdin-capable". Or even stdin-only which is flexible enough:
```
journalctl -k | tools/sof-ipc-timer.py
tools/sof-ipc-timer.py < capture.txt
```

```python
for line in sys.stdin:
```
``

Reply:
@marc-hb currently the file is read twice (at least AFAICS), wouldn't be very convenient with `stdin`